Applications/Spire/Include/Spire/Ui/QuantityInputWidget.hpp

```diff
@@ -1,48 +1,61 @@
 #ifndef SPIRE_QUANTITY_INPUT_WIDGET_HPP
 #define SPIRE_QUANTITY_INPUT_WIDGET_HPP
 #include "Nexus/Definitions/Quantity.hpp"
-#include "Spire/Ui/CustomQtVariants.hpp"
-#include "Spire/Ui/TextInputWidget.hpp"
-#include "Spire/Ui/Ui.hpp"
+#include "Spire/Ui/DecimalInputWidget.hpp"
 
 namespace Spire {
 
-  //! Displays an input box that accepts Quantities.
-  class QuantityInputWidget : public TextInputWidget {
+  //! Represents a widget for displaying and modifying Quantities.
+  class QuantityInputWidget : public QWidget {
```

Comment:
Have you considered inheriting from ```QAbstractSpinBox```?

Reply:
I did, but it involves defining the valueFromText, textFromValue, and validate functions because the default implementations do nothing, among other tasks to get it to work. I thought a double could represent the types we need so modifying the QDoubleSpinBox would be the most straightforward way to implement this.

Reply:
Okay sounds good.